lib/io.toit

```diff
@@ -421,8 +421,46 @@ abstract class Reader implements old-reader.Reader:
 
       n -= size
       processed_ += buffered_.first.size
-      first-array-position_ = 0
       buffered_.remove-first
+      first-array-position_ = 0
+
+  /**
+  Skips all bytes up to and including the given $delimiter.
+
+  Returns the number of bytes skipped including the $delimiter.
+
+  If $to is given, then the search is limited to the given range.
+  If $throw-if-missing is true and the $delimiter is not in the remaining data, throws.
+  If $throw-if-missing is false and the $delimiter is not in the remaining data, skips
+    all remaining data.
+  */
+  skip-up-to delimiter/int --to/int?=null --throw-if-missing/bool=false -> int:
```

Comment:
I think it would be easier to remember if this was called `--throw-if-absent` here and in the other places it exists.

Reply:
I will change it in a follow-up PR together with the read-up-to.